### PR TITLE
ensure that generated validators are up to date

### PR DESCRIPTION
**Pull request by Uzlopak**
With this PR we ensure that we keep our generated code up date

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — APPROVED** [review]
LGTM. 
I hope there would be a Github Action that can be schedule run and take care of it.

File: package.json [hunk 1]
```diff
@@ -19,8 +20,9 @@
     "lint:typescript": "eslint -c types/.eslintrc.json types/**/*.d.ts test/types/**/*.test-d.ts",
     "prepublishOnly": "PREPUBLISH=true tap --no-check-coverage test/build/**.test.js",
     "test": "npm run lint && npm run unit && npm run test:typescript",
-    "test:ci": "npm run unit -- -R terse --cov --coverage-report=lcovonly && npm run test:typescript",
+    "test:ci": "test:validator:integrity && npm run unit -- -R terse --cov --coverage-report=lcovonly && npm run test:typescript",
```
Comment (climba03003) [inline package.json:23]:
```suggestion
    "test:ci": "npm run test:validator:integrity && npm run unit -- -R terse --cov --coverage-report=lcovonly && npm run test:typescript",
```

**Comment by Uzlopak** [comment]
hmm. strange. it fails

```yaml
name: ci

on:
  schedule:
    - cron: 0 8 * * *
  workflow_dispatch:

jobs:
  build-integrity:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        id: checkout

      - name: Use Node.js
        id: setup-node
        uses: actions/setup-node@v3
        with:
          node-version: "lts/*"

      - name: Install
        id: install
        run: |
          npm install --ignore-scripts
      
      - name: Check Integrity
        id: check
        continue-on-error: true
        run: |
          npm run test:validator:integrity

      - name: Create branch if necessary
        run: |
          git config user.name 'github-actions[bot]'
          git config user.email 'github-actions[bot]@users.noreply.github.com'
          git checkout -b rebuild-validators
          git add . &&
          git commit -m "Rebuild Validators"
          git push origin HEAD:rebuild-validators

      - name: Create PR if necessary
        if: steps.check.outcome != 'success'
        uses: octokit/request-action@v2.x
        with:
          route: POST /repos/{owner}/{repo}/pulls
          owner: fastify
          repo: fastify
          base: main
          head: rebuild-validators
          title: Rebuilt Validators
          body: The generated validators are rebuilt.
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
CI seems failing

File: package.json [hunk 2]
```diff
@@ -19,8 +20,9 @@
     "lint:typescript": "eslint -c types/.eslintrc.json types/**/*.d.ts test/types/**/*.test-d.ts",
     "prepublishOnly": "PREPUBLISH=true tap --no-check-coverage test/build/**.test.js",
     "test": "npm run lint && npm run unit && npm run test:typescript",
-    "test:ci": "npm run unit -- -R terse --cov --coverage-report=lcovonly && npm run test:typescript",
+    "test:ci": "npm run test:validator:integrity && npm run unit -- -R terse --cov --coverage-report=lcovonly && npm run test:typescript",
     "test:report": "npm run lint && npm run unit:report && npm run test:typescript",
+    "test:validator:integrity": "npm run build:validation && git diff --quiet lib/error-serializer.js && git diff --quiet lib/configValidator.js",
```
Comment (Eomm) [inline package.json:25]:
```suggestion
    "test:validator:integrity": "npm run build:validation && git diff --quiet --ignore-all-space --ignore-blank-lines --ignore-cr-at-eol lib/error-serializer.js && git diff --quiet --ignore-all-space --ignore-blank-lines --ignore-cr-at-eol lib/configValidator.js",
```

**Review by jsumners — DISMISSED** [review]

File: lib/error-serializer.js [hunk 3]
```diff
@@ -5,6 +5,9 @@
 
   
 
+// eslint-disable-next-line
+const STR_ESCAPE = /[\u0000-\u001f\u0022\u005c\ud800-\udfff]|[\ud800-\udbff](?![\udc00-\udfff])|(?:[^\ud800-\udbff]|^)[\udc00-\udfff]/
```
[the remark this answers is not in the document]
Comment (Eomm) [reply lib/error-serializer.js:9]: This is a generated file, so we rely on ajv here

**Comment by mcollina** [comment]
There seem to be a problem on Windows.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
I would actually move this check to the pre-publish stage and not run it on CI.

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**